packages/flutter/test/widgets/widget_inspector_test.dart

```diff
@@ -2381,6 +2716,500 @@ class _TestWidgetInspectorService extends TestWidgetInspectorService {
+        (WidgetTester tester) async {
+          final Widget widget = Directionality(
+            textDirection: TextDirection.ltr,
+            child: Stack(
+              children: const <Widget>[
+                Text('a'),
+                Text('b', textDirection: TextDirection.ltr),
+                Text('c', textDirection: TextDirection.ltr),
+              ],
+            ),
+          );
+          await tester.pumpWidget(widget);
+          final Element elementA = find.text('a').evaluate().first;
+          service.setSelection(elementA, 'my-group');
+
+          service.testExtension('addPubRootDirectories', <String, String>{
+            'arg0': pubRootTest,
+            'isolateId': '34',
+          });
+          service.testExtension('addPubRootDirectories', <String, String>{
+            'arg0': '/invalid/$pubRootTest',
+            'isolateId': '34',
+          });
+          expect(
+            await service.testExtension(
+              'getSelectedWidget',
+              <String, String>{'objectGroup': 'my-group', 'isolateId': '34',},
+            ),
+            contains('createdByLocalProject'),
+          );
+        },
+      );
+      testWidgets(
+        'can handle removing an unrelated pubRootDirectory',
+        (WidgetTester tester) async {
+          final Widget widget = Directionality(
+            textDirection: TextDirection.ltr,
+            child: Stack(
+              children: const <Widget>[
+                Text('a'),
+                Text('b', textDirection: TextDirection.ltr),
+                Text('c', textDirection: TextDirection.ltr),
+              ],
+            ),
+          );
+          await tester.pumpWidget(widget);
+          final Element elementA = find.text('a').evaluate().first;
+          service.setSelection(elementA, 'my-group');
+
+          service.testExtension('addPubRootDirectories', <String, String>{
+            'arg0': pubRootTest,
+            'arg1': '/invalid/$pubRootTest',
+            'isolateId': '34',
+          });
+          expect(
+            await service.testExtension(
+              'getSelectedWidget',
+              <String, String>{'objectGroup': 'my-group', 'isolateId': '34',},
+            ),
+            contains('createdByLocalProject'),
+          );
+
+          service.testExtension('removePubRootDirectories', <String, String>{
+            'arg0': '/invalid/$pubRootTest',
+            'isolateId': '34',
+          });
+          expect(
+            await service.testExtension(
+              'getSelectedWidget',
+              <String, String>{'objectGroup': 'my-group', 'isolateId': '34',},
+            ),
+            contains('createdByLocalProject'),
+          );
+        },
+      );
+    },
+      skip: !WidgetInspectorService.instance.isWidgetCreationTracked(), // [intended] Test requires --track-widget-creation flag.
+    );
+
     testWidgets('ext.flutter.inspector.trackRebuildDirtyWidgets', (WidgetTester tester) async {
```

Comment:
I don't know why the diff chooses to show this part as replaced.
You can see it unchanged at the bottom of the green replacement text.